pandas/tests/test_index.py

```diff
@@ -2852,6 +2852,18 @@ def test_constructor_corner(self):
         with tm.assertRaisesRegexp(TypeError, 'casting'):
             Int64Index(arr_with_floats)
 
+    def test_notimplementederror_shift_int64(self):
```

Comment:
move the ``NotImplementedErrorr`` case to the ``Base`` class so its tested on all index types; while in the datetimelike override